### DISCUSSION TITLE
Syntax for shortening deconstruct.

### DISCUSSION THREAD

**Discussion by MkazemAkhgary** [dotnet/csharplang#942] [General]
The syntax for deconstruct is like this

```
var (x, y) = obj;
```

its good, but you cant inline these

```
public void Foo(object x, object y) {}
//...
var (x, y) = obj;
Foo(x, y);
//...
var (x, y) = obj;
return (x, y);
```

following are ways to allow deconstruct in various cases.

**Deconstruct in method arguments:**

```
public void Foo(var(object x, object y))
{
     // able to use x and y directly.
}
//...
Foo(obj);
```

**Deconstruct in method returns**

```
public var(object x, object y) Foo()
{
     return obj; // deconstructs into valueTuple(x, y)
}
```

**Remark**

`var(x, y)` is basically extractor of value tuple such as `(? x, ? y) t`. therefor tuple could be also passed in method with `var(? x, ? y)` as argument. and can also be used as return value.

```
(int x, int y) t = (3,3);
var(x, y) = t;
```

**Comment by yaakov-h** [comment]
The first case of deconstructing when calling a method sounds similar to #424.

**Comment by MkazemAkhgary** [comment]
@yaakov-h `..` syntax is interesting too, it also avoids requirement of changing methods signature.

**Comment by orthoxerox** [comment]
See also #153 for the first case.
